primary/src/grpc_server/validator.rs

```diff
@@ -73,12 +65,11 @@ impl<PublicKey: VerifyingKey, KeyMapper: PublicKeyMapper<PublicKey>>
     /// parsed public key. The Err() will hold a Status message with the
     /// specific error description.
     fn get_public_key(&self, request: RoundsRequest) -> Result<PublicKey, Status> {
-        let key =
-            self.public_key_mapper
-                .map(request.public_key.ok_or_else(|| {
-                    Status::invalid_argument("Invalid public key: no key provided")
-                })?)
-                .map_err(|_| Status::invalid_argument("Invalid public key: couldn't parse"))?;
+        let proto_key = request
+            .public_key
+            .ok_or_else(|| Status::invalid_argument("Invalid public key: no key provided"))?;
+        let key = PublicKey::from_bytes(proto_key.bytes.as_ref())
```

Comment:
Ah, completely missed this trait implementation for the `PublicKey` ! This is great...my java instincts worked more in this case and followed the `mapper` path. Thanks @huitseeker for this, much nicer!